src/pybind/mgr/orchestrator.py

```diff
@@ -0,0 +1,266 @@
+
+"""
+ceph-mgr orchestrator interface
+
+This is a DRAFT for discussion.
+
+Goal: enable UI workflows for cluster service management
+      (such as creating OSDs, in addition to stateless services)
+      using common concepts that are implemented by
+      diverse backends such as Rook, DeepSea, ceph-ansible
+
+Concepts:
+    "Stateful service": a daemon that uses local storage, such as OSD or mon.
+    "Stateless service": a daemon that doesn't use any local storage, such
+                         as an MDS, RGW, nfs-ganesha, iSCSI gateway.
+    "Label": arbitrary string tags that may be applied by administrators
+             to nodes.  Typically administrators use labels to indicate
+             which nodes
```

Comment:
This sentence seems to be incomplete. Would it make sense to add that "labels" might be called differently in other deployment tools, e.g. "roles" in DeepSea?

Reply:
Oh, I see - a "label" can be used to mark which roles a specific node *could* assume, it does not necessarily mean it already posesses that role?

Reply:
Right -- the label is just something the human admin indicated, it doesn't mean that the node is physically configured in any particular way.